cgi-bin/LJ/Sysban.pm

```diff
@@ -209,6 +209,12 @@ sub sysban_check {
         return 0 unless scalar @domains >= 2;
         my $domain = "$domains[-2].$domains[-1]";
         return 1 if $check->('email_domain', $domain);
+        # also check for three-element domain bans
```

Comment:
Yes, the rules as I understood them (possibly wrong):

* If "foo.com" is specified, ban it and any/all subdomains of "foo.com" (three elements, ten elements, whatever)
* If "a.foo.com" is specified, ban it specifically

I.e. there is a short form (broad ban) and a long form (specific ban). After reading @alierak's comment I agree that probably the better rule is simply "a ban of a domain includes that domain and all subdomains", which is clearer. Although slightly more annoying to implement, and my solution certainly won't work there.

@kareila Let's not to go down a path of writing 50 lines of code because each line is "simple" and thinking that 50 lines of simple code is going to be "more maintainable" than 5 lines of more complex code. I would strongly disagree on that matter.